src/main/java/ch/njol/skript/bukkitutil/EnchantmentUtils.java

```diff
@@ -18,26 +18,165 @@
  */
 package ch.njol.skript.bukkitutil;
 
+import ch.njol.skript.classes.ClassInfo;
+import ch.njol.skript.classes.Parser;
+import ch.njol.skript.classes.Serializer;
+import ch.njol.skript.lang.ParseContext;
+import ch.njol.skript.localization.Language;
+import ch.njol.util.StringUtils;
+import ch.njol.yggdrasil.Fields;
 import org.bukkit.NamespacedKey;
+import org.bukkit.Registry;
 import org.bukkit.enchantments.Enchantment;
-import org.eclipse.jdt.annotation.Nullable;
+import org.jetbrains.annotations.Nullable;
 
-import com.google.common.collect.BiMap;
-import com.google.common.collect.HashBiMap;
-import ch.njol.skript.Skript;
+import java.io.StreamCorruptedException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
 
 /**
  * Maps enchantments to their keys.
  */
 public class EnchantmentUtils {
 
+	private static final Map<Enchantment, String> NAMES = new HashMap<>();
+	private static final Map<String, Enchantment> PATTERNS = new HashMap<>();
+	private static final boolean HAS_REGISTRY = BukkitUtils.registryExists("ENCHANTMENT");
+
+	static {
+		if (!HAS_REGISTRY) {
+			Language.addListener(() -> {
+				NAMES.clear();
+				PATTERNS.clear();
+				for (Enchantment enchantment : Enchantment.values()) {
+					NamespacedKey key = enchantment.getKey();
+					final String[] names = Language.getList("enchantments." + key.getKey());
+
+					if (!names[0].startsWith("enchantments.")) {
+						NAMES.put(enchantment, names[0]);
+						// Add lang file names
+						for (String name : names)
+							PATTERNS.put(name.toLowerCase(Locale.ENGLISH), enchantment);
+					}
+					// If Minecraft provided, add key without namespace and underscores (ex: "fire aspect")
+					if (key.getNamespace().equalsIgnoreCase("minecraft"))
```

Comment:
```suggestion
					if (key.getNamespace().equalsIgnoreCase(NamespacedKey.MINECRAFT))
```

Reply:
I could have SWORN we went over this and I did that ... must have been the registry PR

Reply:
yep it was